config/jobs/kubernetes-csi/gen-jobs.sh

```diff
@@ -288,7 +280,7 @@ snapshotter_version() {
         # All other jobs test against the latest stable snapshotter
         # release. Additional jobs could be created to cover version
         # skew, if desired.
-        echo '"v3.0.2"'
+        echo '"v3.0.3"'
```

Comment:
hm now we have 4.0 but it requires 1.20 as a min version cc @xing-yang

Reply:
I suppose we need a switch statement somewhere then?

Reply:
Yeah, maybe set it to 4.0 for 1.20 and higher and set it to 3.0.3 for 1.19 or lower.

Reply:
So we don't need (or want) to test 3.0.3 on 1.20? Some users might run it there.

That's fine with me, I just wanted to be sure.

Reply:
Added a Kubernetes check.